### PR TITLE
Incremented version to 2.4.2.

### PR DESCRIPTION
**Pull request by dreamer-89**
Manual update on autocut PR https://github.com/opensearch-project/OpenSearch/pull/5558. This PR adds the 1.3.8 bwcVersion after 1.3.7 release

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .ci/bwcVersions [hunk 1]
```diff
@@ -55,3 +55,4 @@ BWC_VERSION:
   - "2.3.0"
```
Comment (reta) [inline .ci/bwcVersions:55]: @dreamer-89 we need 1.3.8 here as well (if I am not mistaken)

Comment (dreamer-89) [reply .ci/bwcVersions:56]: Thanks @reta . Updated.

**Review by reta — APPROVED** [review]

**Comment by dreamer-89** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/8076/
> * **CommitID:** [baca918](https://github.com/opensearch-project/OpenSearch/commit/baca918f6ed5a1ad0ee6256801ba8777a8b0b34d)
>   Please examine the workflow log, locate, and copy-paste the failure below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

Jave heap space erorr. Refiring!
```
Failed to execute org.gradle.cache.internal.AsyncCacheAccessDecoratedCache$$Lambda$413/0x0000000801034000@8015504.
java.lang.OutOfMemoryError: Java heap space
Failed to execute org.gradle.cache.internal.AsyncCacheAccessDecoratedCache$$Lambda$413/0x0000000801034000@39c3bb40.
java.lang.OutOfMemoryError: Java heap space
```

**Comment by reta** [comment]
> > ### Gradle Check (Jenkins) Run Completed with:
> > 
> > * **RESULT:** FAILURE ❌
> > * **URL:** https://build.ci.opensearch.org/job/gradle-check/8076/
> > * **CommitID:** [baca918](https://github.com/opensearch-project/OpenSearch/commit/baca918f6ed5a1ad0ee6256801ba8777a8b0b34d)
> >   Please examine the workflow log, locate, and copy-paste the failure below, then iterate to green.
> >   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?
> 
> Jave heap space erorr. Refiring!
> 
> ```
> Failed to execute org.gradle.cache.internal.AsyncCacheAccessDecoratedCache$$Lambda$413/0x0000000801034000@8015504.
> java.lang.OutOfMemoryError: Java heap space
> Failed to execute org.gradle.cache.internal.AsyncCacheAccessDecoratedCache$$Lambda$413/0x0000000801034000@39c3bb40.
> java.lang.OutOfMemoryError: Java heap space
> ```

https://github.com/gradle/gradle/issues/1413 apparently :(

**Comment by dreamer-89** [comment]
> > > ### Gradle Check (Jenkins) Run Completed with:
> > > 
> > > * **RESULT:** FAILURE ❌
> > > * **URL:** https://build.ci.opensearch.org/job/gradle-check/8076/
> > > * **CommitID:** [baca918](https://github.com/opensearch-project/OpenSearch/commit/baca918f6ed5a1ad0ee6256801ba8777a8b0b34d)
> > >   Please examine the workflow log, locate, and copy-paste the failure below, then iterate to green.
> > >   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?
> > 
> > 
> > Jave heap space erorr. Refiring!
> > ```
> > Failed to execute org.gradle.cache.internal.AsyncCacheAccessDecoratedCache$$Lambda$413/0x0000000801034000@8015504.
> > java.lang.OutOfMemoryError: Java heap space
> > Failed to execute org.gradle.cache.internal.AsyncCacheAccessDecoratedCache$$Lambda$413/0x0000000801034000@39c3bb40.
> > java.lang.OutOfMemoryError: Java heap space
> > ```
> 
> [gradle/gradle#1413](https://github.com/gradle/gradle/issues/1413) apparently :(

Do we need to open one on our end for tracking, I have seen few builds recently failing with this ?

**Comment by reta** [comment]
> > > > ### Gradle Check (Jenkins) Run Completed with:
> > > > 
> > > > * **RESULT:** FAILURE x
> > > > * **URL:** https://build.ci.opensearch.org/job/gradle-check/8076/
> > > > * **CommitID:** [baca918](https://github.com/opensearch-project/OpenSearch/commit/baca918f6ed5a1ad0ee6256801ba8777a8b0b34d)
> > > >   Please examine the workflow log, locate, and copy-paste the failure below, then iterate to green.
> > > >   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?
> > > 
> > > 
> > > Jave heap space erorr. Refiring!
> > > ```
> > > Failed to execute org.gradle.cache.internal.AsyncCacheAccessDecoratedCache$$Lambda$413/0x0000000801034000@8015504.
> > > java.lang.OutOfMemoryError: Java heap space
> > > Failed to execute org.gradle.cache.internal.AsyncCacheAccessDecoratedCache$$Lambda$413/0x0000000801034000@39c3bb40.
> > > java.lang.OutOfMemoryError: Java heap space
> > > ```
> > 
> > 
> > [gradle/gradle#1413](https://github.com/gradle/gradle/issues/1413) apparently :(
> 
> Do we need to open one on our end for tracking, I have seen few builds recently failing with this ?

We kind of had this one https://github.com/opensearch-project/OpenSearch/issues/3973 but it seems like there is a multitude of the OOM flavors ...

**Pull request merged**